lib/rules/no-mutation-props.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Prevent direct mutation of this.props
+ * @author Ian Schmitz
+ */
+'use strict';
+
+var has = require('has');
+var Components = require('../util/Components');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent direct mutation of this.props',
+      category: 'Possible Errors',
+      recommended: true
+    }
+  },
+
+  create: Components.detect(function (context, components, utils) {
+
+    /**
+     * Checks if the component is valid
+     * @param {Object} component The component to process
+     * @returns {Boolean} True if the component is valid, false if not.
+     */
+    function isValid(component) {
+      return Boolean(component && !component.mutateProps);
```

Comment:
let's use `‼x` instead of `Boolean(x)` to coerce to booleans - and in this case, `‼component && !component.mutateProps` is sufficient